src/js/services/optimizely.test.js

```diff
@@ -0,0 +1,241 @@
+import OptimizelyClient, {
+  storeExperimentParticipation,
+  isExperimentAlreadyViewed,
+  trackExperimentViewed,
+  STORAGE_KEY,
+  COOKIE_KEY,
+} from './optimizely';
+import AnalyticsClient from '../services/analytics.js';
+import glob from '../../../jest/global';
+
+import { default as CookieOrginal } from 'js-cookie';
+jest.mock('js-cookie');
+const Cookie = CookieOrginal;
+
+describe('Optimizely Service', () => {
+  const client = new OptimizelyClient();
+  const userDataReady = new CustomEvent('userDataReady');
+
+  const orgId = 'circle';
+  const experimentKey = 'experiment_key';
+  const variationName = 'variation_name';
+
+  const options = {
+    organizationId: '00000000-0000-0000-0000-000000000000',
+    userAnalyticsId: '11111111-1111-1111-1111-111111111111',
+    experimentKey: 'experiment_key',
+    groupExperimentName: 'experiment_group_test',
+    experimentContainer: 'experiment_container',
+  };
+
+  beforeEach(() => {
+    // ensure local storage clear
+    window.localStorage.clear();
+  });
+
+  afterEach(() => {
+    glob.userData = null;
+    glob.forceAll = null;
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+  });
+
+  describe('getUserId()', () => {
+    it('test if userData exists but analytics_id undefined', () => {
```

Comment:
usually, test names should describe what they are expecting rather than testing. For example for this one that would be: `returns null when userData is empty`